File: CS/pdf-viewer-sticky-note/Form1.cs
using DevExpress.Pdf;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Linq;


namespace pdf_viewer_sticky_note
{
    public partial class Form1 : RibbonForm
    {
        public Form1()
        {
            InitializeComponent();
            pdfViewer.LoadDocument("Demo.pdf");
        }
        static Matrix CreateTransform(int degree, PdfRectangle cropBox)
        {
            Matrix rotationMatrix;
            switch (degree % 360)
            {
                case 90:
                    rotationMatrix = new Matrix(0, -1, 1, 0, 0, (float)cropBox.Width);
                    break;
                case 180:
                    rotationMatrix = new Matrix(-1, 0, 0, -1, (float)cropBox.Width, (float)cropBox.Height);
                    break;
                case 270:
                    rotationMatrix = new Matrix(0, 1, -1, 0, (float)cropBox.Height, 0);
                    break;
                default:
                    rotationMatrix = new Matrix(1, 0, 0, 1, 0, 0);
                    break;
            }
            var matrix = new Matrix(1, 0, 0, 1, (float)-cropBox.Left, (float)-cropBox.Bottom);
            using (rotationMatrix)
                matrix.Multiply(rotationMatrix, MatrixOrder.Append);
            matrix.Invert();
            return matrix;
        }
        static int NormalizeRotate(int rotate)
        {
            int step = -360;
            if (rotate < 0)
                step = 360;
            while ((rotate < 0 && step > 0) || (rotate > 270 && step < 0))
                rotate += step;
            return rotate;
        }
        void AddStickyNote()
        {
            var pageInfo = pdfViewer.GetPageInfo(1);
            int normalizedAngle = NormalizeRotate(pdfViewer.RotationAngle + pageInfo.RotationAngle);
            var cropBox = pageInfo.CropBox;
            PdfRectangle viewCropBox;
            if (normalizedAngle == 90 || normalizedAngle == 270)
                viewCropBox = new PdfRectangle(0, 0, cropBox.Height, cropBox.Width);
            else
                viewCropBox = new PdfRectangle(0, 0, cropBox.Width, cropBox.Height);
            PdfPoint stickyNotePoint = new PdfPoint(viewCropBox.Right - 20, viewCropBox.Top - 20);
            var matrix = CreateTransform(normalizedAngle, cropBox);
            var points = new[] { new PointF((float)stickyNotePoint.X, (float)stickyNotePoint.Y) };
            matrix.TransformPoints(points);
            PdfDocumentPosition stickyNotePosition = new PdfDocumentPosition(1, new PdfPoint(points[0].X, points[0].Y));
            pdfViewer.AddStickyNote(stickyNotePosition, "Note", Color.Yellow);
        }

        private void barButtonItem1_ItemClick(object sender, ItemClickEventArgs e)
        {
            AddStickyNote();
        }
    }
}

File: VB/pdf-viewer-sticky-note/Form1.vb
Imports DevExpress.Pdf
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon
Imports System.Drawing
Imports System.Drawing.Drawing2D

Namespace pdf_viewer_sticky_note

    Public Partial Class Form1
        Inherits RibbonForm

        Public Sub New()
            InitializeComponent()
            pdfViewer.LoadDocument("Demo.pdf")
        End Sub

        Private Shared Function CreateTransform(ByVal degree As Integer, ByVal cropBox As PdfRectangle) As Matrix
            Dim rotationMatrix As Matrix
            Select Case degree Mod 360
                Case 90
                    rotationMatrix = New Matrix(0, -1, 1, 0, 0, CSng(cropBox.Width))
                Case 180
                    rotationMatrix = New Matrix(-1, 0, 0, -1, CSng(cropBox.Width), CSng(cropBox.Height))
                Case 270
                    rotationMatrix = New Matrix(0, 1, -1, 0, CSng(cropBox.Height), 0)
                Case Else
                    rotationMatrix = New Matrix(1, 0, 0, 1, 0, 0)
            End Select

            Dim matrix = New Matrix(1, 0, 0, 1, CSng(-cropBox.Left), CSng(-cropBox.Bottom))
            Using rotationMatrix
                matrix.Multiply(rotationMatrix, MatrixOrder.Append)
            End Using

            matrix.Invert()
            Return matrix
        End Function

        Private Shared Function NormalizeRotate(ByVal rotate As Integer) As Integer
            Dim [step] As Integer = -360
            If rotate < 0 Then [step] = 360
            While rotate < 0 AndAlso [step] > 0 OrElse rotate > 270 AndAlso [step] < 0
                rotate += [step]
            End While

            Return rotate
        End Function

        Private Sub AddStickyNote()
            Dim pageInfo = pdfViewer.GetPageInfo(1)
            Dim normalizedAngle As Integer = NormalizeRotate(pdfViewer.RotationAngle + pageInfo.RotationAngle)
            Dim cropBox = pageInfo.CropBox
            Dim viewCropBox As PdfRectangle
            If normalizedAngle = 90 OrElse normalizedAngle = 270 Then
                viewCropBox = New PdfRectangle(0, 0, cropBox.Height, cropBox.Width)
            Else
                viewCropBox = New PdfRectangle(0, 0, cropBox.Width, cropBox.Height)
            End If

            Dim stickyNotePoint As PdfPoint = New PdfPoint(viewCropBox.Right - 20, viewCropBox.Top - 20)
            Dim matrix = CreateTransform(normalizedAngle, cropBox)
            Dim points = {New PointF(CSng(stickyNotePoint.X), CSng(stickyNotePoint.Y))}
            matrix.TransformPoints(points)
            Dim stickyNotePosition As PdfDocumentPosition = New PdfDocumentPosition(1, New PdfPoint(points(0).X, points(0).Y))
            pdfViewer.AddStickyNote(stickyNotePosition, "Note", Color.Yellow)
        End Sub

        Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            AddStickyNote()
        End Sub
    End Class
End Namespace
